libs/src/services/discoveryProvider/index.js

```diff
@@ -676,11 +684,20 @@ class DiscoveryProvider {
       await this.autoSelectEndpoint()
```

Comment:
Chatted w/ @piazz about this one, let's make default retries = 3 going forward since we have this nice fallback logic now. Making it 10 might just make you keep trying the same discprov over and over again w/o anything working and slows down the time until we switch to something healthy.